platform/source/minimal-printf/mbed_printf_implementation.c

```diff
@@ -261,6 +265,25 @@ static void mbed_minimal_formatted_string_void_pointer(char *buffer, size_t leng
  */
 static void mbed_minimal_formatted_string_double(char *buffer, size_t length, int *result, double value, FILE *stream)
 {
+    /* check for nan */
+    if (isnan(value)) {
+        mbed_minimal_formatted_string_string(buffer, length, result, "nan", 3, stream);
+        return;
+    }
+
+    /* check for inf */
+    if (isinf(value)) {
+        /* print positive or negative */
+        if (isinf(value) < 0) {
```

Comment:
In C+11, `isinf()`  [returns a bool](http://www.cplusplus.com/reference/cmath/isinf/?kw=isinf).
I am not sure you can do this to decide whether to print `-` or `+`.

Reply:
You just need to do `if (value < 0)`.